docs/src/test-api/class-testconfig.md

```diff
@@ -44,13 +44,17 @@ export default defineConfig({
     - `threshold` ?<[float]> an acceptable perceived color difference between the same pixel in compared images, ranging from `0` (strict) and `1` (lax). `"pixelmatch"` comparator computes color difference in [YIQ color space](https://en.wikipedia.org/wiki/YIQ) and defaults `threshold` value to `0.2`.
     - `maxDiffPixels` ?<[int]> an acceptable amount of pixels that could be different, unset by default.
     - `maxDiffPixelRatio` ?<[float]> an acceptable ratio of pixels that are different to the total amount of pixels, between `0` and `1` , unset by default.
+    - `maxDiffSize` ?<[int]> an acceptable number of pixels to vary the width or height by, `1` by default.
```

Comment:
Let's make it work with existing `maxDiffPixels` and `maxDiffPixelRatio` options instead of introducing two new parameters. Let's consider those pixels which are missing on the images as different, as if you padded the images (from right and bottom) to make the sizes match.

Let's start with addressing exact rounding error problem. We can allow image dimensions to differ by up to 1px in each direction, if the difference is within specified tolerance thresholds, they will be equal. This should cover (hopefully) common cases where the screenshot differ due to rounding error. So far, we haven't seen pictures of legit cases where the difference would be more than 1px in each dimension.

Reply:
Since this change is significantly different, I have started a fresh PR here:
https://github.com/microsoft/playwright/pull/28453